ibis/backends/duckdb/__init__.py

```diff
@@ -633,3 +633,23 @@ def drop_view(
 
         with self.begin() as con:
             con.execute(view)
+
+    def _insert_dataframe(
+        self,
+        table_name: str,
+        df: pd.DataFrame,
```

Comment:
How does it work without using `df` in the function body?

Reply:
DuckDB inspect variables in the outer scope.

Reply:
It's not ideal, but otherwise we have to generate another view which requires another pass over the data for duckdb to infer object columns' data types and that's visible in the benchmark.

Reply:
Actually, I don't know what duckdb is doing when `con.register` is called (part of the `duckdb_engine` API) but it does lose some performance.

Reply:
> DuckDB inspects variables in the outer scope.

Can you add a comment about this since it's not straightforward.

Reply:
Done!